[Review] feat & fix: 리뷰 생성 기능 검증 로직 수정 & 테스트 케이스 전체 추가

rlqor9688 opened the pull request:
## Description
- ReviewServiceImpl 클래스 수정: 검증 로직 순서를 변경하였습니다. (requestDto 필드값 DB 존재 여부 확인 > 자격 검증 > 유효성 검증 순)
- ReviewServiceImplTest 클래스 수정: ReviewServiceImpl 클래스 내 createReview 메서드에서 발생하는 모든 예외 상황에 대한 테스트케이스를 추가하였습니다.

## Changes
0. createReview_리뷰_생성에_성공한다() 내부 입력 인자 수정
1. createReview_authUser가_null이면_USER_NOT_FOUND_예외를_던진다
2. createReview_유저가_없으면_USER_NOT_FOUND_예외를_던진다()
3. createReview_주문이_없으면_ORDER_NOT_FOUND_에러를_던진다()
4. createReview_가게가_없으면_STORE_NOT_FOUND_예외를_던진다()
5. createReview_유저상태가_INACTIVE면_DELETED_USER_예외를_던진다()
6. createReview_유저권한이_CONSUMER가_아니면_ROLE_CONSUMER_FORBIDDEN_예외를_던진다()
7. createReview_본인주문이_아니면_FORBIDDEN_예외를_던진다()
8. createReview_주문상태가_COMPLETE가_아니면_REVIEW_ORDER_INVALID_예외를_던진다()
9. createReview_주문상품이_없으면_REVIEW_ORDERITEM_NOT_FOUND_예외를_던진다()
10. createReview_주문과_가게가_매칭되지_않으면_REVIEW_STORE_MISMATCH_예외를_던진다()

## Screenshots
![image](https://github.com/user-attachments/assets/d278dac5-0845-4f50-8b78-ce5a0b1989cb)

rlqor9688 committed:
feat [review] : ReviewServiceImplTest 테스트케이스 추가

0. createReview_리뷰_생성에_성공한다() 내부 입력 인자 수정

1. createReview_authUser가_null이면_USER_NOT_FOUND_예외를_던진다

2. createReview_유저가_없으면_USER_NOT_FOUND_예외를_던진다()

3. createReview_주문이_없으면_ORDER_NOT_FOUND_에러를_던진다()

4. createReview_가게가_없으면_STORE_NOT_FOUND_예외를_던진다()

5. createReview_유저상태가_INACTIVE면_DELETED_USER_예외를_던진다()

6. createReview_유저권한이_CONSUMER가_아니면_ROLE_CONSUMER_FORBIDDEN_예외를_던진다()

7. createReview_본인주문이_아니면_FORBIDDEN_예외를_던진다()

8. createReview_주문상태가_COMPLETE가_아니면_REVIEW_ORDER_INVALID_예외를_던진다()

9. createReview_주문상품이_없으면_REVIEW_ORDERITEM_NOT_FOUND_예외를_던진다()

10. createReview_주문과_가게가_매칭되지_않으면_REVIEW_STORE_MISMATCH_예외를_던진다()

rlqor9688 committed:
feat [review] : ReviewServiceImplTest 테스트케이스 추가

0. createReview_리뷰_생성에_성공한다() 내부 입력 인자 수정

1. createReview_authUser가_null이면_USER_NOT_FOUND_예외를_던진다

2. createReview_유저가_없으면_USER_NOT_FOUND_예외를_던진다()

3. createReview_주문이_없으면_ORDER_NOT_FOUND_에러를_던진다()

4. createReview_가게가_없으면_STORE_NOT_FOUND_예외를_던진다()

5. createReview_유저상태가_INACTIVE면_DELETED_USER_예외를_던진다()

6. createReview_유저권한이_CONSUMER가_아니면_ROLE_CONSUMER_FORBIDDEN_예외를_던진다()

7. createReview_본인주문이_아니면_FORBIDDEN_예외를_던진다()

8. createReview_주문상태가_COMPLETE가_아니면_REVIEW_ORDER_INVALID_예외를_던진다()

9. createReview_주문상품이_없으면_REVIEW_ORDERITEM_NOT_FOUND_예외를_던진다()

10. createReview_주문과_가게가_매칭되지_않으면_REVIEW_STORE_MISMATCH_예외를_던진다()

rlqor9688 committed:
feat [review] : ReviewServiceImplTest 테스트케이스 추가

0. createReview_리뷰_생성에_성공한다() 내부 입력 인자 수정

1. createReview_authUser가_null이면_USER_NOT_FOUND_예외를_던진다

2. createReview_유저가_없으면_USER_NOT_FOUND_예외를_던진다()

3. createReview_주문이_없으면_ORDER_NOT_FOUND_에러를_던진다()

4. createReview_가게가_없으면_STORE_NOT_FOUND_예외를_던진다()

5. createReview_유저상태가_INACTIVE면_DELETED_USER_예외를_던진다()

6. createReview_유저권한이_CONSUMER가_아니면_ROLE_CONSUMER_FORBIDDEN_예외를_던진다()

7. createReview_본인주문이_아니면_FORBIDDEN_예외를_던진다()

8. createReview_주문상태가_COMPLETE가_아니면_REVIEW_ORDER_INVALID_예외를_던진다()

9. createReview_주문상품이_없으면_REVIEW_ORDERITEM_NOT_FOUND_예외를_던진다()

10. createReview_주문과_가게가_매칭되지_않으면_REVIEW_STORE_MISMATCH_예외를_던진다()

rlqor9688 committed:
fix [review] : ReviewRequestDto 내 rating 검증 어노테이션 수정

1. ReviewRequestDto 내 rating의 어노테이션을 @NotBlank에서 @NotNull로 수정하였습니다.

2. messages.properties에 notnull 메시지로 변경하였습니다.

rlqor9688 committed:
Merge branch 'dev' into feature/kiback-review

anyubinn commented on src/main/java/com/delivery/igo/igo_delivery/api/review/service/ReviewServiceImpl.java:
```diff
@@ -49,29 +47,28 @@ public ReviewResponseDto createReview(AuthUser authUser, ReviewRequestDto reques
 
+        // 유효한 유저인지 검증(UserStatus= LIVE, UserRole = CCONSUMER)
+        findUser.validateDelete();
+        findUser.validateConsumer();
```
유저 유효성 검증을 사용자 조회 직후에 바로 수행하면 이후 로직 전에 비정상적인 요청을 빠르게 차단할 수 있어 전체 흐름상 더 안전할 것 같아요!

rlqor9688 replied:
감사합니다! 수정사항 반영했습니다!

rlqor9688 committed:
Merge branch 'dev' into feature/kiback-review

rlqor9688 committed:
refactoring [review] : ReviewServiceImpl 유효성 검증 순서  , ReviewServiceImplTest 데이터 수정

1. ReviewServiceImpl 내 유효성 검증 순서를 수정하였습니다.

2. ReviewServiceImplTest 순서를 수정한 유효성 검증 순서에 맞춰 변경하고, 입력 데이터를 수정하였습니다.
